plugins/tomcat8/src/main/java/org/radargun/service/TomcatServerService.java

```diff
@@ -85,31 +85,34 @@ public void init() {
       log.info("Java home: " + java);
 
       lifecycle = new TomcatServerLifecycle(this);
-      try {
-         URL resource = getClass().getResource("/" + file);
-         Path filesystemFile = FileSystems.getDefault().getPath(file);
-         Path target = FileSystems.getDefault().getPath(catalinaBase, "conf", "radargun-tomcat-" + ServiceHelper.getContext().getSlaveIndex() + ".xml");
-         if (resource != null) {
-            try (InputStream is = resource.openStream()) {
-               log.info("Found " + file + " as a resource");
-               Files.copy(is, target, StandardCopyOption.REPLACE_EXISTING);
+
+      evaluateFile((evaluatedFile) -> {
```

Comment:
Once again, I wonder if there is a way to remove some of this boiler plate code? Maybe a method in the `ProcessService` class?

Reply:
When https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R61 is overwritten each class has a different behavior.

https://github.com/radargun/radargun/pull/584/files#diff-58c2bbab38ba23ee91c39b662dc67ec3R89 is the custom code for tomcat

https://github.com/radargun/radargun/pull/584/files#diff-3fb0eecefe0cce27cfd6a152bf6e968aR105 is the custom code for Infinispan

The boiler plate code is https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R120

When we don't overwrite https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R61 there is no additional code and `ProcessService` will do the transformation. See https://github.com/radargun/radargun/pull/584/files#diff-096483ddb920163d93ac6499fb4e8320R66